client/my-sites/checkout/src/lib/get-flow-plan-features.ts

```diff
@@ -2,13 +2,118 @@ import {
 	applyTestFiltersToPlansList,
 	getPlan,
 	FEATURE_CUSTOM_DOMAIN,
+	IncompleteWPcomPlan,
 } from '@automattic/calypso-products';
+import {
+	NEWSLETTER_FLOW,
+	isLinkInBioFlow,
+	isAnyHostingFlow,
+	isNewsletterOrLinkInBioFlow,
+	isBlogOnboardingFlow,
+} from '@automattic/onboarding';
 import { ResponseCartProduct } from '@automattic/shopping-cart';
 import { getPlanFeaturesObject } from 'calypso/lib/plans/features-list';
-import {
-	getHighlightedFeatures,
-	getPlanFeatureAccessor,
-} from 'calypso/my-sites/plan-features/util';
+
+const newsletterFeatures = ( flowName: string, plan: IncompleteWPcomPlan ) => {
+	return flowName === NEWSLETTER_FLOW && plan.getNewsletterSignupFeatures;
+};
+
+const linkInBioFeatures = ( flowName: string, plan: IncompleteWPcomPlan ) => {
+	return isLinkInBioFlow( flowName ) && plan.getLinkInBioSignupFeatures;
+};
+
+const hostingFeatures = ( flowName: string, plan: IncompleteWPcomPlan ) => {
+	return isAnyHostingFlow( flowName ) && plan.getHostingSignupFeatures?.( plan.term );
+};
+
+const blogOnboardingFeatures = ( flowName: string, plan: IncompleteWPcomPlan ) => {
+	return isBlogOnboardingFlow( flowName ) && plan.getBlogOnboardingSignupFeatures;
+};
+
+const signupFlowDefaultFeatures = (
+	flowName: string,
+	plan: IncompleteWPcomPlan,
+	isInVerticalScrollingPlansExperiment: boolean
+) => {
+	if ( ! flowName || isNewsletterOrLinkInBioFlow( flowName ) ) {
+		return;
+	}
+
+	return isInVerticalScrollingPlansExperiment
+		? plan.getSignupFeatures
+		: plan.getSignupCompareAvailableFeatures;
+};
+
+const getPlanFeatureAccessor = ( {
+	flowName = '',
+	isInVerticalScrollingPlansExperiment = false,
```

Comment:
Can we remove the `isInVerticalScrollingPlansExperiment` prop? It is passed as `false` always:
https://github.com/Automattic/wp-calypso/blob/2289bd822fcadee173e49003a57a412fec6b08f3/client/my-sites/checkout/src/lib/get-flow-plan-features.ts#L41

Reply:
Hey @aneeshd16 I am gradually boxing these changes to reduce widespread impact. My current commit order is shown here.
https://github.com/Automattic/wp-calypso/issues/82335#issuecomment-1781940845

Will be diving more aggressively to cleanup props once these are committed :)

Reply:
Might have to explore this a bit.
https://github.com/Automattic/wp-calypso/blob/a001b3b00068b47ae9ee125502a55b02a72050e6/client/signup/steps/plans/index.jsx#L350-L353